src/Simulation/EntryPointDriver/Driver.cs

```diff
@@ -13,66 +13,161 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
-using Microsoft.Quantum.Simulation.Core;
-using static Microsoft.Quantum.EntryPointDriver.Driver;
 
 namespace Microsoft.Quantum.EntryPointDriver
 {
+    using Validators = List<ValidateSymbol<CommandResult>>;
```

Comment:
I don't think you're actually using the mutability of `List` - `ImmutableList` might be better.

Reply:
My preference is to use `List` here as that is more in-line with the mutable-by-default way C# works.

Reply:
Regardless of C#'s original design, it does have immutable collections now, and I'd recommend using them when possible unless mutation is needed, because immutable data structures are significantly easier to reason about. That would also be consistent with the rest of the entry point driver code which uses `ImmutableList`.

Reply:
I disagree in general with what I see as an over-use of immutables in a language like C# that has mutable variables as a default, and I don't believe using immutables makes it easier for developers such as myself to reason about the code (actually it is quite the opposite, there have been many times working in F# where I have wanted to pull my hair out over the lack of mutables 😄). I suspect the reason you may find immutables easier to reason with is because of your background with functional programming, but perhaps I am wrong?

Regardless, I can see your point that, despite my objections, immutable data types are everywhere in this code base, so it would be more consistent to have this be one as well. I will update this to an `ImmutableList`.

Reply:
There's a lot of discussion in the industry about the advantages of immutable data ([here's an example](https://onor.io/2014/05/12/what-is-immutability-and-why-should-i-care/)), so I don't think it's just me. Using immutable types rules out an entire class of hard-to-fix bugs (state being accidentally mutated by a distant part of the program). They also have some disadvantages, so I don't think they should be used everywhere, but I think using mutable data types should be a conscious choice when mutability is specifically needed, and not the default option.

Reply:
changed to ImmutableList